lib/std/mem.zig

```diff
@@ -1111,21 +1096,20 @@ pub fn indexOf(comptime T: type, haystack: []const T, needle: []const T) ?usize
 /// more sophisticated algorithm on larger inputs.
 pub fn lastIndexOfLinear(comptime T: type, haystack: []const T, needle: []const T) ?usize {
     var i: usize = haystack.len - needle.len;
-    while (true) : (i -= 1) {
-        if (mem.eql(T, haystack[i .. i + needle.len], needle)) return i;
-        if (i == 0) return null;
-    }
+    return while (true) : (i -= 1) {
+        if (eql(T, haystack[i .. i + needle.len], needle)) break i;
```

Comment:
This isn't the first or last of these, but this kind of change - which this PR seems in large part focused on - seems to have no real reasoning behind it, it's just using language features for the sake of using language features. Using C-style early returns is more readable, easier to optimize, and takes up fewer characters: I see no reason for these changes.

Reply:
> it's just using language features for the sake of using language features.

That was one of my points of changing them but i have reverted them back.